tests/server/blueprints/institutes/test_institute_views.py

```diff
@@ -1,7 +1,314 @@
+    # GIVEN an institute with a phenotype model having no subpanels
+    store.create_phenomodel(institute_obj["internal_id"], "Test model", "Model description")
+    model_obj = store.phenomodel_collection.find_one()
+    assert model_obj["subpanels"] == {}
+
+    # GIVEN an initialized app
+    # GIVEN a valid user and institute
+    with app.test_client() as client:
+        # GIVEN that the user could be logged in
+        resp = client.get(url_for("auto_login"))
+
+        form_data = dict(
+            title="Subpanel title",
+            subtitle="Subpanel subtitle",
+            add_subpanel="save subpanel",
+        )
+        # WHEN the user creates subpanel in phenotype model via POST request
+        resp = client.post(
+            url_for(
+                "overview.phenomodel",
+                institute_id=institute_obj["internal_id"],
+                model_id=model_obj["_id"],
+            ),
+            data=form_data,
+        )
+        # Then the subpanel dictionary should be added to model subpanels
+        updated_model = store.phenomodel_collection.find_one()
+        subpanel_id = list(updated_model["subpanels"].keys())[0]
+        assert updated_model["subpanels"][subpanel_id]["title"] == "Subpanel title"
+        assert updated_model["subpanels"][subpanel_id]["subtitle"] == "Subpanel subtitle"
+
+        # WHEN the user sends a POST request to remove the subpanel
+        form_data = dict(subpanel_delete=subpanel_id)
+        resp = client.post(
+            url_for(
+                "overview.phenomodel",
+                institute_id=institute_obj["internal_id"],
+                model_id=model_obj["_id"],
+            ),
+            data=form_data,
+        )
+        # THEN the model should be removed from models subpanels
+        updated_model = store.phenomodel_collection.find_one()
+        assert updated_model["subpanels"] == {}
+
+
+def test_phenomodel_POST_add_omim_checkbox_to_subpanel(app, user_obj, institute_obj):
+    """Test adding an OMIM checkbox to a subpanel of a phenotype model via POST request"""
+
+    # GIVEN an institute with a phenotype model
+    store.create_phenomodel(institute_obj["internal_id"], "Test model", "Model description")
+    model_obj = store.phenomodel_collection.find_one()
+    # containing a subpanel
+    model_obj["subpanels"] = {"subpanel_x": TEST_SUBPANEL}
+    store.update_phenomodel(model_obj["_id"], model_obj)
+    model_obj = store.phenomodel_collection.find_one()
+    assert model_obj["subpanels"]["subpanel_x"]
+
+    # GIVEN that database contains the HPO term to add to the subopanel
+    store.disease_term_collection.insert_one(OMIM_TERM)
+
+    # GIVEN an initialized app
+    # GIVEN a valid user and institute
+    with app.test_client() as client:
+        resp = client.get(url_for("auto_login"))
+
+        # WHEN the user creates an OMIM checkbox using the endpoint
+        form_data = dict(
```

Comment:
Fixture?

Reply:
Just a part of this, because the form data in this case is using the omim checkbox, the other form_data is using an hpo checkbox. I'll use the checkboxes fixtures inside the form_data instead!